### PR TITLE
adding pipeline fixes

### PR DESCRIPTION
**Pull request by gcampbell-msft**

### CODE REVIEW

**Comment by gcampbell-msft** [comment]
There's still an issue going on with the pipeline, but I think I've got the authorization working at least.

**Comment by gcampbell-msft** [comment]
There seems to be an issue with publishing due to a 404 and it not finding the URL, I think this has more to do with npm or something being down there, rather than this pipeline, I think I have the pipeline completely confirmed. 

I will continue to test locally until the publishing issue is resolved, and once it is, we can merge this PR and see if it publishes!

File: azure-pipelines.yml
Comment (benmcmorran) [inline azure-pipelines.yml:66]: What does "self" mean here and how is it different than the default behavior?

File: jobs/shared/build.yml
Comment (benmcmorran) [inline jobs/shared/build.yml:3]: If you want to break these all into separate job/task files I don't mind, but it's probably not really necessary for such a small build.

**Review by benmcmorran — APPROVED** [review]

File: jobs/shared/build.yml
Comment (gcampbell-msft) [reply jobs/shared/build.yml:3]: I felt like I might as well, because I needed to share the jobs.

File: .npmrc
Comment (gcampbell-msft) [reply .npmrc:1]: I was copying a pattern from my user .npmrc, it's likely not necessary. I'll take it out and assuming it works, we'll leave it out.

File: azure-pipelines.yml
Comment (gcampbell-msft) [reply azure-pipelines.yml:60]: Removing comment. I'll leave the TeamName for robustness.

File: azure-pipelines.yml
Comment (gcampbell-msft) [reply azure-pipelines.yml:58]: Will my powershell script work inside the linux-based machine?

File: azure-pipelines.yml
Comment (gcampbell-msft) [reply azure-pipelines.yml:66]: I believe it has to do with "syncing sources". Basically, I believe that this in combination with the persist credentials is what is used to tell the pipeline that we need to be able to modify the repo or at least run git scripts on it. 

https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines#examples

Comment (benmcmorran) [reply azure-pipelines.yml:65]: Thanks for clarifying.

File: azure-pipelines.yml
Comment (benmcmorran) [reply azure-pipelines.yml:58]: Yep, `pwsh` should be installed on the Linux agents.

**Review by benmcmorran — DISMISSED** [review]

File: azure-pipelines.yml
Comment (gcampbell-msft) [reply azure-pipelines.yml:58]: Perfect, I'll update the slashes and the name and then I'll update the package.json and then merge.

**Pull request merged**
